synapse/lib/remcycle.py

```diff
@@ -0,0 +1,846 @@
+        '''
+        try:
+            self._parseConfig(config=config, reload_config=reload_config)
+        except Exception as e:
+            log.exception('Failed to process configuration')
+            raise e
+
+    def _parseConfig(self, config, reload_config):
+
+        for key in self.required_keys:
+            if key not in config:
+                log.error('Remcycle config is missing a required value %s.', key)
+                raise NoSuchName('Missing required key.')
+
+        _apis = config.get(APIS)
+        _namespace = config.get(NAMESPACE)
+        _doc = config.get(DOC)
+
+        if _namespace in self.namespaces:
+            if reload_config:
+                self.delConfig(namespace=_namespace)
+            else:
+                raise NameError('Namespace is already registered.')
+
+        self.docs[_namespace] = _doc
+        self.global_request_headers[_namespace] = {k: v for k,v in config.get(HTTP, {}).items()}
+
+        # Register APIs
+        for varn, val in _apis.items():
+            name = ':'.join([_namespace, varn])
+            # Stamp api http config ontop of global config, then stamp it into the API config
+            _http = self.global_request_headers[_namespace].copy()
+            _http.update(val.get('http', {}))
+            val['http'] = _http
+            nyx_obj = Nyx(config=val)
+            self._register_api(name=name, obj=nyx_obj)
+        self.namespaces.add(_namespace)
+
+        self.fire('hypnos:register:namespace:add', namespace=_namespace)
+
+    def _register_api(self, name, obj):
+        '''
+        Register a Nyx object and any corresponding ingest definitions to the
+        cortex.
+        '''
+        if name in self.apis:
+            raise NameError('Already registered {}'.format(name))
+        self.apis[name] = obj
+        for gest_name, gest in obj.gests.items():
+            action_name = ':'.join([name, gest_name])
+            # Register the action with the attached cortex
+            ingest_func = s_ingest.register_ingest(core=self.core,
+                                                   gest=gest,
+                                                   evtname=action_name,
+                                                   ret_func=True
+                                                   )
+
+            def gest_glue(event):
+                evtname, event_args = event
+                kwargs = event_args.get('kwargs')
+                resp = kwargs.get('resp')
+                data = resp.get('data')
+                self.core.fire(action_name, data=data)
+
+            # Register the action to unpack the async.Boss job results and fire the cortex event
+            self.on(name=name, func=gest_glue)
+
+            # Store things for later reuse (for deregistartion)
+            self._api_ingests[name].append((action_name, ingest_func, gest_glue))
+
+        self.fire('hypnos:register:api:add', api=name)
+
+    def delConfig(self, namespace):
```

Comment:
Renamed everything in Hypnos to be "Web" as a prefix, since that is short and easy to type.